image/v1/types.go

```diff
@@ -267,8 +290,25 @@ type TagImportPolicy struct {
 	Insecure bool `json:"insecure,omitempty" protobuf:"varint,1,opt,name=insecure"`
 	// Scheduled indicates to the server that this tag should be periodically checked to ensure it is up to date, and imported
 	Scheduled bool `json:"scheduled,omitempty" protobuf:"varint,2,opt,name=scheduled"`
+	// ImportMode describes how to import an image manifest.
+	ImportMode ImportModeType `json:"importMode,omitempty" protobuf:"bytes,3,opt,name=importMode,casttype=ImportModeType"`
 }
 
+// ImportModeType describes how to import an image manifest.
+type ImportModeType string
+
+const (
```

Comment:
these are inscrutable to me, but I couldn't say what's better. :(

Reply:
well that isn't great. we tried to be clear in the docs, did that not help? I'm up for trying to improve that at least - the names were hard to pick :s